ops/__init__.py

```diff
@@ -12,31 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""The ops library: a Python framework for writing Juju charms.
+"""The API to respond to Juju events and manage the application.
 
-The ops library is a Python framework (`available on PyPI`_) for developing
-and testing Juju charms in a consistent way, using standard Python constructs
-to allow for clean, maintainable, and reusable code.
+This module provides:
 
-A charm is an operator -- business logic encapsulated in a reusable software
-package that automates every aspect of an application's life.
+- :class:`~ops.CharmBase` and :class:`~ops.Object` to register the charm or
+  a charm lib, respectively, with the framework.
+- [OPT1] :class:`~ops.framework.EventBase` class and individual event types, like
+  :class:`~ops.ActionEvent` class.
+- [OPT2] :class:`~ops.framework.EventBase` and its subclasses:
 
-Charms written with ops support Kubernetes using Juju's "sidecar charm"
-pattern, as well as charms that deploy to Linux-based machines and containers.
+  - :class:`~ops.HookEvent` class for Juju events, like the
+    :class:`~ops.StartEvent` class.
+  - :class:`~ops.LifecycleEvent` class for synthetic events, like the
+    :class:`~ops.CollectStatusEvent` class.
```

Comment:
@tmihoc we've discussed this at the standup, and have decided to go with OPT1 for now, for these reasons:

1. charmers should consume events and ideally not care about the distinction between Juju and Ops events; we certainly don't want charmers to "depend" on ops implementation details
2. exposing current, admittedly somewhat broken/misnamed class hierarchy doesn't feel right; happy to revisit this later if we decide to clean up "lifecycle" and "what is a juju event" tangles. In any case, out of scope of this PR.